client/components/AuthForm.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+
+import { useRegisterMutation } from "../features/api";
+
+
+const AuthForm = () => {
+
+    const [register, { isLoading, isError, data }] = useRegisterMutation();
+    const [email, setEmail] = useState("");
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [confirmPassword, setConfirmPassword] = useState("");
+  
+    const handleSubmit = (e) => {
+      e.preventDefault();
+      register({ email, username, password });
```

Comment:
Not sure if you saw my previous comments, about moving the register mutation up a level into the RegisterPage and passing onSubmit callback into the AuthForm? Not blocking for merging this PR, just want to make sure it's tracked, since it'll make it easier to make this AuthForm reusable for Login (could be added at the same time that you build the login page though)

Reply:
Added it as an Issue to address later